src/commands/file_ops.rs

```diff
@@ -49,45 +49,92 @@ pub fn paste(context: &mut AppContext, options: IoWorkerOptions) -> JoshutoResul
 }
 
 pub fn copy_filename(context: &mut AppContext) -> JoshutoResult<()> {
-    let entry_file_name = match context
+    let entry_file_name = context
         .tab_context_ref()
         .curr_tab_ref()
         .curr_list_ref()
         .and_then(|c| c.curr_entry_ref())
-    {
-        Some(entry) => Some(entry.file_name().to_string()),
-        None => None,
-    };
+        .and_then(|entry| Some(entry.file_name().to_string()));
+
     if let Some(file_name) = entry_file_name {
-        let clipboards = [
-            (
-                "wl-copy",
-                format!("printf '%s' {} | {} 2> /dev/null", file_name, "wl-copy"),
-            ),
-            (
-                "xsel",
-                format!("printf '%s' {} | {} -ib 2> /dev/null", file_name, "xsel"),
-            ),
-            (
-                "xclip",
-                format!(
-                    "printf '%s' {} | {} -selection clipboard 2> /dev/null",
-                    file_name, "xclip"
-                ),
+        copy_string_to_buffer(file_name)?;
+    }
+    Ok(())
+}
+
+pub fn copy_filepath(context: &mut AppContext) -> JoshutoResult<()> {
+    let entry_file_path = context
+        .tab_context_ref()
+        .curr_tab_ref()
+        .curr_list_ref()
+        .and_then(|c| c.curr_entry_ref())
+        .and_then(|entry| entry.file_path().to_str())
+        .and_then(|s| Some(s.to_string()));
+
+    if let Some(file_path) = entry_file_path {
+        copy_string_to_buffer(file_path)?;
+    }
+    Ok(())
+}
+
+pub fn copy_dir(context: &mut AppContext) -> JoshutoResult<()> {
+    let opt_entry = context
+        .tab_context_ref()
+        .curr_tab_ref()
+        .curr_list_ref()
+        .and_then(|c| c.curr_entry_ref());
+
+    let dir:  Option<String> = opt_entry
```

Comment:
A bit confused on what this is trying to do:
let's say we have `/home/user/file.txt`, would this copy `/home/user` into the clipboard?
If so, you can use:
```rust
context
    .tab_context_ref()
    .curr_tab_ref()
    .curr_list_ref().file_path();
```

Reply:
Thank you! This is really simpler!